doc/tutorials/imgproc/imgtrans/distance_transformation/distance_transform.markdown

```diff
@@ -0,0 +1,165 @@
+Image Segmentation with Distance Transform and Watershed Algorithm {#tutorial_distance_transform}
+=============
+
+Goal
+----
+
+In this tutorial you will learn how to:
+
+-   Use the OpenCV function @ref cv::filter2D in order to perform some laplacian filtering for image sharpening
+-   Use the OpenCV function @ref cv::distanceTransform in order to obtain the derived representation of a binary image, where the value of each pixel is replaced by its distance to the nearest background pixel
+-   Use the OpenCV function @ref cv::watershed in order to isolate objects in the image from the background
+
+Theory
+------
+
+Code
+----
+
+This tutorial code's is shown lines below. You can also download it from
+    [here](https://github.com/Itseez/opencv/tree/master/samples/cpp/tutorial_code/ImgTrans/imageSegmentation.cpp).
+@includelineno samples/cpp/tutorial_code/ImgTrans/imageSegmentation.cpp
+
+Explanation / Result
+--------------------
+
+-#  Load the source image and check if it is loaded without any problem, then show it:
+    @code{.cpp}
```

Comment:
Since this and all following code snippets are taken directly from example file, it is better to use `snippet` doxygen command.
You can read about it here: http://docs.opencv.org/ref/master/d4/db1/tutorial_documentation.html#tutorial_documentation_commands_include
And more details here:
http://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdsnippet